### ISSUE TITLE
Flow mode supports server TLS

### ISSUE THREAD

**Issue by dtrejod** [grafana/agent#4252]
### Request

In grafana-agent flow, there doesn't appear to be any support for configuring HTTP server/GRPC TLS.
https://grafana.com/docs/agent/v0.34/flow/reference/cli/run/

TLS was supported in static mode via the[ `-server.http.enable-tls`](https://grafana.com/docs/agent/v0.34/static/configuration/flags/#server) CLI flag. The server would then reference the [`server_config` ](https://grafana.com/docs/agent/v0.34/static/configuration/server-config/#server_tls_config) block to pull in the relevant TLS config settings. 

### Use case

I can imagine this impacts a number of use cases:
- If a user wants to enable grafana-agent [clustering](https://grafana.com/docs/agent/v0.34/flow/reference/cli/run/#clustered-mode-experimental) with flow, it would need to be done over insecure protocols. 
- Wanting the flow UI to be served with (m)TLS. 
- Wanting to collect [`Component Controller` ](https://grafana.com/docs/agent/v0.34/flow/monitoring/controller_metrics/#controller-metrics) or other grafana-agent internal metrics from the grafana-agent service itself over https.

**Comment by rfratto** [comment]
Hi, this is a duplicate of #2715. I'd like to get #2715 implemented in the near future, but it's blocked by finding a way to cleanly allow TLS to be configured in the config file, which is more challenging in Flow thanks to the ability to reference components. 

I've been actively working on a proposal to unblock #2715, and I hope to share more information about it soon.

Closing as a duplicate, please feel free to continue the discussion in #2715!
